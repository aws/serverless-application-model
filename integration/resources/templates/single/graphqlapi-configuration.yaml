Transform: AWS::Serverless-2016-10-31
Resources:
  SuperCoolAPI:
    Type: AWS::Serverless::GraphQLApi
    Properties:
      SchemaInline: |
        type Book {
            bookName: String
            id: ID
        } 
        type Query { getBook(bookName: String): Book }
      OwnerContact: blah-blah
      Auth:
        Type: API_KEY
      ApiKeys:
        MyApiKey: {}
      Functions:
        processQuery:
          Runtime:
            Name: APPSYNC_JS
            Version: 1.0.0
          DataSource: NONE
          InlineCode: |
            import { util } from '@aws-appsync/utils';

            export function request(ctx) {
              const id = util.autoId();
              return { payload: { ...ctx.args, id } };
            }

            export function response(ctx) {
              return ctx.result;
            }
      Resolvers:
        Query:
          getBook:
            Pipeline:
            - processQuery

  IntrospectionDisableSuperCoolAPI:
    Type: AWS::Serverless::GraphQLApi
    Properties:
      SchemaInline: |
        type Book {
            bookName: String
            id: ID
        } 
        type Query { getBook(bookName: String): Book }
      OwnerContact: blah-blah
      IntrospectionConfig: DISABLED
      QueryDepthLimit: 10
      ResolverCountLimit: 100
      Auth:
        Type: API_KEY
      ApiKeys:
        MyApiKey: {}
      Functions:
        processQuery:
          Runtime:
            Name: APPSYNC_JS
            Version: 1.0.0
          DataSource: NONE
          InlineCode: |
            import { util } from '@aws-appsync/utils';

            export function request(ctx) {
              const id = util.autoId();
              return { payload: { ...ctx.args, id } };
            }

            export function response(ctx) {
              return ctx.result;
            }
      Resolvers:
        Query:
          getBook:
            Pipeline:
            - processQuery
Outputs:
  SuperCoolAPI:
    Description: AppSync API
    Value: !GetAtt SuperCoolAPI.GraphQLUrl
  MyApiKey:
    Description: API Id
    Value: !GetAtt SuperCoolAPIMyApiKey.ApiKey
  IntrospectionDisableSuperCoolAPI:
    Description: AppSync API
    Value: !GetAtt IntrospectionDisableSuperCoolAPI.GraphQLUrl
  IntrospectionDisableSuperCoolAPIMyApiKey:
    Description: API Id
    Value: !GetAtt IntrospectionDisableSuperCoolAPIMyApiKey.ApiKey

Metadata:
  SamTransformTest: true
