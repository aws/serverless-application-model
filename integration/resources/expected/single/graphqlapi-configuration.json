[
  {
    "LogicalResourceId": "SuperCoolAPI",
    "ResourceType": "AWS::AppSync::GraphQLApi"
  },
  {
    "LogicalResourceId": "SuperCoolAPICloudWatchRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "SuperCoolAPISchema",
    "ResourceType": "AWS::AppSync::GraphQLSchema"
  },
  {
    "LogicalResourceId": "SuperCoolAPIQuerygetBook",
    "ResourceType": "AWS::AppSync::Resolver"
  },
  {
    "LogicalResourceId": "SuperCoolAPINoneDataSource",
    "ResourceType": "AWS::AppSync::DataSource"
  },
  {
    "LogicalResourceId": "SuperCoolAPIprocessQuery",
    "ResourceType": "AWS::AppSync::FunctionConfiguration"
  },
  {
    "LogicalResourceId": "SuperCoolAPIMyApiKey",
    "ResourceType": "AWS::AppSync::ApiKey"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPI",
    "ResourceType": "AWS::AppSync::GraphQLApi"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPICloudWatchRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPISchema",
    "ResourceType": "AWS::AppSync::GraphQLSchema"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPIQuerygetBook",
    "ResourceType": "AWS::AppSync::Resolver"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPINoneDataSource",
    "ResourceType": "AWS::AppSync::DataSource"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPIprocessQuery",
    "ResourceType": "AWS::AppSync::FunctionConfiguration"
  },
  {
    "LogicalResourceId": "IntrospectionDisableSuperCoolAPIMyApiKey",
    "ResourceType": "AWS::AppSync::ApiKey"
  }
]
